### PR TITLE
`find()` fails on YAML sequences

### PR DESCRIPTION
**Pull request by akarve**
Some pandas args take lists  (sequences in YAML). `build.py:find` did not work for this case; it would descend the children of the sequence and then blow up because, e.g. in test case below, the children (ints) are not iterable.

```
subgroup:
   # transform and skiprows apply to all children, unless overidden
   transform: tsv
   skiprows: [0,5,20]
```

### CODE REVIEW

**Comment by akarve** [comment]
@asah this is a proper diff of the fix; yes I checked for callers

File: quilt/tools/build.py [hunk 1]
```diff
@@ -198,16 +198,20 @@ def build_package(username, package, yaml_path, checks_path=None, dry_run=False,
...
-            if k == key:
-                yield v
-            elif isinstance(v, dict):
-                for result in find(key, v):
-                    yield result
-            elif isinstance(v, list):
-                for item in v:
-                    for result in find(key, item):
+        try:
+            vals = iteritems(value)
```
Comment (dimaryaz) [inline quilt/tools/build.py]: Leave it as it was, a part of the `for ... in`. Storing the iterator is kind of weird.

File: quilt/tools/build.py [hunk 2]
```diff
@@ -198,16 +198,20 @@ def build_package(username, package, yaml_path, checks_path=None, dry_run=False,
...
+                if k == key:
+                    yield v
+                elif isinstance(v, dict):
+                    for result in find(key, v):
                         yield result
+                elif isinstance(v, list):
+                    for item in v:
+                        for result in find(key, item):
+                            yield result
+        except AttributeError:
```
Comment (dimaryaz) [inline quilt/tools/build.py]: This feels hacky, and can hide actual errors. Check if the object is an instance of `collections.Iterable`?

File: quilt/tools/build.py [hunk 3]
```diff
@@ -198,17 +198,21 @@ def build_package(username, package, yaml_path, checks_path=None, dry_run=False,
...
-                for result in find(key, v):
-                    yield result
-            elif isinstance(v, list):
-                for item in v:
-                    for result in find(key, item):
+        """
+        find matching nodes recursively;
+        only descend iterables
+        """
+        if(isinstance(value, Iterable)):
```
Comment (dimaryaz) [inline quilt/tools/build.py]: Remove parens and add a space: `if isinstance(...):`

**Pull request merged**